x-pack/test/functional/services/cases/files.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export function CasesFilesTableServiceProvider({ getService, getPageObject }: FtrProviderContext) {
+  const common = getPageObject('common');
+  const testSubjects = getService('testSubjects');
+  const find = getService('find');
+  const browser = getService('browser');
+
+  const assertFileExists = (index: number, totalFiles: number) => {
+    if (index > totalFiles - 1) {
+      throw new Error('Cannot get file from table. Index is greater than the length of all rows');
+    }
+  };
+
+  return {
+    async addFile(fileInputPath: string) {
+      // click the AddFile button
+      await testSubjects.click('cases-files-add');
+      await find.byCssSelector('[aria-label="Upload a file"]');
+
+      // upload a file
+      await common.setFileInputPath(fileInputPath);
+      await testSubjects.click('uploadButton');
+    },
+
+    async searchByFileName(fileName: string) {
+      const searchField = await testSubjects.find('cases-files-search');
+
+      searchField.clearValue();
+
+      await searchField.type(fileName);
+      await searchField.pressKeys(browser.keys.ENTER);
+    },
+
+    async deleteFile(index: number = 0) {
+      const row = await this.getFileByIndex(index);
+
+      (await row.findByCssSelector('[data-test-subj="cases-files-delete-button"]')).click();
+
+      await testSubjects.click('confirmModalConfirmButton');
```

Comment:
Maybe we should await for the modal to appear before trying to click the confirm button.

Reply:
`testSubjects.click` does a find that has a default time out. Should I do another `find` on line 46? With a custom timeout?

Reply:
If there is already a timeout in the click then we are fine. The flaky test runner should reveal any flakiness.